main/api/src/mill/api/ClassLoader.scala

```diff
@@ -8,25 +8,43 @@ import io.github.retronym.java9rtexport.Export
 import scala.util.Try
 
 object ClassLoader {
+
   def java9OrAbove = !System.getProperty("java.specification.version").startsWith("1.")
+
+  @deprecated("Use other create method instead.", "mill after 0.9.7")
+  def create(urls: Seq[URL],
+             parent: java.lang.ClassLoader,
+             sharedLoader: java.lang.ClassLoader,
+             sharedPrefixes: Seq[String])
+            (implicit ctx: Ctx.Home): URLClassLoader = {
+    create(
+      urls = urls,
+      parent = parent,
+      sharedLoader = sharedLoader,
+      sharedPrefixes = sharedPrefixes,
+      logger = None)
+  }
+
   def create(urls: Seq[URL],
              parent: java.lang.ClassLoader,
              sharedLoader: java.lang.ClassLoader = getClass.getClassLoader,
```

Comment:
Since this default argument is the dangerous part, should this be deprecated too?
I expect the deprecated method have this default value so the old code use the deprecated method and get the deprecation notice. Since you're defaulting the logger to `None` I think the old code is using this method directly.
Can you explain what's going on here?

Reply:
The changed method is source-compatible to the old one, because the new argument (the logger) has a default value. But it's not binary compatible, so we need an overload with the old signature. But in Scala only one overload can have default arguments, that's why the deprecated method no longer has default arguments, but the new one.

The default value for the `sharedLoader` is ok as long as you don't need to share classes not already visible to mill core. This is true for in most cases (e.g. sbt-test-interface, jna, scala{_,js,native} worker api), but not in case of playlib.

The motivation to change this method was solely the wish to log the shared classloading when using mill with `--debug` option.

Reply:
But you're probable right, we should take the opportunity to remove the default for the classloader, as even if it works now, it may hinder us later, when we want to change packaging, embedding or JVM.

Reply:
Yeah, it is a big footgun and can't be checked by our tests which is even scarier. But removing it would change the behaviour of other modules using it, unless you expose a deprecated `classloader` here for the other modules to consume.
As usual, a decision important enough to be tackled in a separate PR; as you said the deprecation here is only about the logger.